### ISSUE TITLE
Improve train-top5.py to target top 10 leaderboard ranking

### ISSUE THREAD

**Issue by mikeycdavis** [mikeycdavis/Numerai#33]
## Context

Current leaderboard position: **#129 / 572** staked data scientists (ranked by MMC20).
Target: **top 10** (requires MMC20 ≥ ~0.0049; #1 currently at 0.0098).

Multipliers are set to **0.75×CORR20v2 + 2.25×MMC**, meaning MMC is the primary payout driver. All improvements should prioritise maximising MMC (unique signal not already in Numerai's meta model).

Analysis informed by reviewing the #1 ranked account (BGUBERFAIN) who runs deliberate ablation experiments across model variants.

---

### 1. Meta model AND benchmark are blended directly into predictions (critical)
```python
preds["meta_model"] = meta_model.reindex(live_features.index)
preds["benchmark"] = live_benchmark_models...mean(axis=1)
ensemble = ((model_ensemble + ranked["meta_model"] + ranked["benchmark"]) / 3)
```
MMC rewards predictions that are *different* from the meta model. Blending the meta model in as an equal vote directly suppresses MMC. Worse, **the benchmark models have a negative cumulative MMC curve over the past year** (visible on BGUBERFAIN's account chart — the gray benchmark line trends steadily below zero). We are blending in predictions that actively drag our MMC down. **Remove both from the ensemble.**

### 2. Using the `small` feature set
```python
feature_cols = feature_metadata["feature_sets"]["small"]
```
Small (~50 features) limits signal. The competitive standard is `medium` (~700 features). However, BGUBERFAIN's ablation shows their FULL feature set model (MMC20=0.0032) underperforms their standard model (MMC20=0.0099) — so `medium` should be tested before jumping to `large`. `colsample_bytree=0.1` should also be tuned upward (0.15–0.2) to match a larger feature set.

### 3. Top-50 neutralization is correct — do not increase it
BGUBERFAIN's ablation confirms: NO50 drops MMC from 0.0099 → 0.0056, N100 drops to 0.0031. **Our existing `[:50]` slice is validated.** More neutralization hurts MMC significantly.

### 4. Limited target diversity
Only 6 targets across 2 families (ender, victor, xerxes, teager2b). More target families = more diverse signal, improving Sharpe and MMC.

### 5. No hyperparameter optimisation
Hyperparameters are hand-tuned. The crypto pipeline already uses Optuna for HPO — the same approach here could meaningfully improve model quality.

### 6. Fold models used directly for live predictions
OOF fold models are trained on 4/5 of the data. A full retrain on 100% of training data after OOF validation would give a stronger live model.

### 7. No ablation experiment framework
BGUBERFAIN runs multiple staked model variants (NO50, N100, FULL, V52) to empirically test changes before committing to them. We have 6 spare models (DJSQUEETZ, DJBEATZ, DJCLEATZ, DJFEATZ, DJTEETZ) that could serve this purpose — each testing one configuration change.

### 8. Linear era-boosting may underweight recent regimes
An exponential ramp would more aggressively prioritise recent market behaviour.

---

## Proposed Changes (priority order)

- [ ] **Remove meta model and benchmark from ensemble blend** — benchmark has *negative* MMC; removing both is the single highest-impact change
- [ ] **Test `medium` feature set** — use a spare model (e.g. DJSQUEETZ) as ablation before making it the default; tune `colsample_bytree` to 0.15–0.2
- [ ] **Keep top-50 neutralization** — validated by BGUBERFAIN ablation; do not increase to 100
- [ ] **Expand target candidates** — add cyrus, ralph, waldo, teager2b_60, and other v5.2 families
- [ ] **Add Optuna HPO** — borrow pattern from `crypto/train-crypto.py` (15–30 trials)
- [ ] **Full retrain after OOF** — use fold models for Sharpe diagnostics, retrain on full data for live predictions
- [ ] **Use spare models as ablation slots** — assign each of DJSQUEETZ/DJBEATZ/etc. to test one config variant per round
- [ ] **Exponential era-boosting** — replace linear ramp with exponential to weight recent eras more aggressively

---

## Reference

- Leaderboard top 10 MMC20 range: 0.0049–0.0098 (572 staked data scientists)
- Current rank: 129/572
- Scoring metric weight: 2.25×MMC, 0.75×CORR20v2
- BGUBERFAIN ablation: standard=0.0099, NO50=0.0056, N100=0.0031, FULL=0.0032
